src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -0,0 +1,201 @@
+        }
+        this.initialLoad = false;
+    }
+
+    /**
+     * Unsubscribes from the subscriptions.
+     */
+    ngOnDestroy() {
+        this._subscriptions.forEach((sub: Subscription) => sub.unsubscribe());
+    }
+
+    /**
+     * Populates the services from the container component.
+     */
+    private populateContainerServices(): void {
+        this._dateIOService = this.getProviderFromContainer(DateIOService);
+        this._dateNavigationService = this.getProviderFromContainer(DateNavigationService);
+        this._datepickerEnabledService = this.getProviderFromContainer(DatepickerEnabledService);
+    }
+
+    /**
+     * Initialize DateIO Subscriptions
+     */
+    private initializeSubscriptions(): void {
+        if (this._dateIOService) {
+            this._subscriptions.push(this._dateIOService.dateStrUpdated.subscribe((dateStr) => {
+                this.writeDateStrToInputField(dateStr);
+                // This makes sure that ngModelChange is fired
+                // TODO: Check if there is a better way to do this.
+                // NOTE: Its important to use NgControl and not NgModel because
+                // NgModel only works with template driven forms
+                if (this._ngControl) {
+                    this._ngControl.control.setValue(dateStr);
+                }
+            }));
+            this._subscriptions.push(this._dateIOService.dateUpdated.subscribe(() => {
+                this._dateUpdated.emit(this._dateIOService.date);
+            }));
+        }
+    }
+
+    /**
+     * Writes the date string value to the input field
+     */
+    private writeDateStrToInputField(value: string): void {
+        this.renderer.setProperty(this.elRef.nativeElement, "value", value);
+    }
+
+    /**
+     * Sets the Input Date on the DateIOService
+     */
+    set inputDate(value: string) {
+        this._dateIOService.inputDate = value;
+    }
+
+    /**
+     * Processes a date object to check if its valid or not.
+     */
+    private processDate(value: Date): void {
+        if (this._dateIOService && value) {
+            // The date object is converted back to string because in Javascript you can create a date object
+            // like this: new Date("Test"). This is a date object but it is invalid. Converting the date object
+            // that the user passed helps us to verify the validity of the date object.
+            this._dateIOService.inputDate = this._dateIOService.toLocaleDisplayFormatString(value);
+            if (this._dateIOService.inputDate && !this.initialLoad) {
+                this.writeDateStrToInputField(this._dateIOService.inputDate);
+            }
+        }
+    }
+
+    private initialLoad: boolean = true;
+    private dateValueOnInitialLoad: Date;
+
+    /**
+     * Javascript Date object input set by the user.
+     */
+    @Input("clrDate")
+    set date(value: Date) {
+        if (this.initialLoad) {
```

Comment:
I'd rather just not have `initialLoad` as an extra property, just testing `if (this._dateIOService)` is enough.